base/cvd/cuttlefish/host/commands/cvd/server_command/start.cpp

```diff
@@ -722,53 +702,40 @@ Result<cvd::Response> CvdStartCommandHandler::Handle(
   if (is_help) {
     Command command =
         CF_EXPECT(ConstructCvdHelpCommand(bin, envs, subcmd_args, request));
-    ShowLaunchCommand(command.Executable(), subcmd_args, envs);
+    ShowLaunchCommand(command, envs);
 
     CF_EXPECT(subprocess_waiter_.Setup(command.Start()));
     auto infop = CF_EXPECT(subprocess_waiter_.Wait());
     return ResponseFromSiginfo(infop);
   }
 
-  auto group_creation_info =
-      CF_EXPECT(GetGroupCreationInfo(bin, subcmd_args, envs, request));
-  Command command =
-      CF_EXPECT(ConstructCvdNonHelpCommand(bin, group_creation_info, request));
+  auto [group, lock_files] = CF_EXPECT(GetGroup(subcmd_args, envs, request));
+  CF_EXPECT(UpdateArgs(subcmd_args, group));
+  CF_EXPECT(UpdateEnvs(envs, group));
+  Command command = CF_EXPECT(
+      ConstructCvdNonHelpCommand(bin, group, subcmd_args, envs, request));
 
   // The instance database needs to be updated if an interrupt is received.
-  auto signals_fd = CF_EXPECT(HandleInterruptSignals());
-  std::thread interrupter_thread([this, signals_fd]() {
-    for (;;) {
-      int signal = 0;
-      auto res = TEMP_FAILURE_RETRY(read(signals_fd, &signal, sizeof(signal)));
-      if (res > 0) {
-        // Interrupt regardless of signal
-        (void)subprocess_waiter_.Interrupt();
-      } else if (res == 0) {
-        // other end closed
-        close(signals_fd);
-        return;
-      } else {
-        auto err = errno;
-        LOG(ERROR) << "Failed to read from signal pipe: " << strerror(err);
-        return;
-      }
-    }
+  auto handle_res = PushInterruptListener([](int) {
+    LOG(WARNING) << "Interrupting signal received, wrapping up";
+    // No need to do anything other than ignore the signal: the signal will be
+    // delivered to the subprocess if it's still running, which will cause it
```

Comment:
Is this assuming the signal is sent to the entire process group? I thought `launch_cvd --daemon` starts its own process group?